### PR TITLE
script: test framework to validate the reproducible builds

### PR DESCRIPTION
**Pull request by ajaykish**
A test framework to verify that builds are reproducible, i.e.
producing exactly the same binary. Validating the binary generated
from same directory, different directories and on different hosts
with same tool chain.

Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>

### CODE REVIEW

File: tests/reproducible_builds/check_reproducible_build.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#!/bin/sh
+
+if [ -d "/home/$USER/zephyr_src1" ]; then
+    rm -rf /home/$USER/zephyr_src1
+fi
+
+if [ -d "/home/$USER/zephyr_src2" ]; then
```
Comment (nashif) [inline tests/reproducible_builds/check_reproducible_build.sh]: /home/$USER = $HOME
and I would not put anything in home, please always use temporary directories with random names in /tmp or something like that.

File: tests/reproducible_builds/check_reproducible_build.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+mkdir /home/$USER/zephyr_src1
+echo "Cloning latest code in zephyr_src1"
+cd /home/$USER/zephyr_src1
+git clone https://github.com/zephyrproject-rtos/zephyr.git
+
+
+
+mkdir /home/$USER/zephyr_src2
+echo "Cloning latest code in zephyr_source2"
```
Comment (nashif) [inline tests/reproducible_builds/check_reproducible_build.sh]: why are we cloning Zephyr twice here? It is enough to build an application outside of Zephyr tree in 2 different location to test this.

File: scripts/check_reproducible_builds.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    res = res1 = res2 = res3 = 0
+    homedir = os.environ['HOME']
+    filename  = os.path.join(homedir, 'out.txt')
+
+    if os.path.exists("filename"):
+        os.remove("filename")
+
+    if os.path.exists(input_file1):
+        if os.path.exists(input_file2):
+            subprocess.call("diffoscope --text filename {0} {1}". format(input_file1, input_file2), shell=True)
```
Comment (nashif) [inline scripts/check_reproducible_builds.py]: you need to add this to python requirements

File: scripts/check_reproducible_builds.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import os
```
Comment (nashif) [inline scripts/check_reproducible_builds.py]: - header with python shebang
- license
- copyright
- quick into into the script, what does it do, how to use it.

**Review by nashif — CHANGES_REQUESTED** [review]
That shell script needs major improvements, it needs to work as a test and cleanly from inside the zephyr tree without touch user homes and duplicating git trees.

Need documentation of what the scripts are doing and how to use them.

**Comment by ajaykish** [comment]
@nashif I have addressed the comments and pushed a new patch.
 Could you please review it.

**Comment by nashif** [comment]
obsoleted, we have other scripts providing this now.

**Comment by nashif** [comment]
#14593

**Pull request closed without merging**